concrete/src/Page/Collection/Collection.php

```diff
@@ -193,7 +193,9 @@ public static function reindexPendingPages()
      */
     public static function createCollection($data)
     {
-        $db = Loader::db();
+        $app = Application::getFacadeApplication();
+        /** @var Connection $db */
+        $db = $app->make(Connection::class);
         $dh = Loader::helper('date');
```

Comment:
you should probably just replace that loader::helper as well

Reply:
Yeah, I can do it, but we need to discuss that.
Refactor whole class, or focus on removing writing queries from executeQuery

Reply:
I think start by fixing just executeQuery. These old classes are so big that it's impossible to review anything if more than one type of change is made at once.